scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala

```diff
@@ -422,8 +422,24 @@ abstract class FixedPathSource(path: String*) extends FileSource {
   // `toString` is used by equals in JobTest, which causes
   // problems due to unstable collection type of `path`
   override def toString = getClass.getName + path.mkString("(", ",", ")")
+  override def hdfsWritePath = stripTrailing(super.hdfsWritePath)
+
   override def hashCode = toString.hashCode
   override def equals(that: Any): Boolean = (that != null) && (that.toString == toString)
+
+  /**
+    * Similar in behavior to {@link TimePathedSource.writePathFor}.
+    * Strip out the trailing slash star.
+    */
+  protected def stripTrailing(path: String): String = {
+    assert(path != "*", "Path must not be *")
```

Comment:
Yeah that might be a good idea as a followup. Ended up choosing assert as the existing code in FileSource & TimePathedSource was already using it and I thought it might be nice to keep it consistent.